### PR TITLE
Create Queens_Problem.go

### LINKED ISSUE #864: N-Queen Problem in cpp,java,go,python and js
Can i start working on it? I know someone is already doing but he may be working on JS implementation so please confirm and say whether i can work on the other one ?

### PR DESCRIPTION
**Pull request by khushal87**
Fixes #864 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

- Added Queens_Problem.go
- 

#### Languages Used:
- GO
- 

#### Files Added:
- Queens_Problem.go
-

### CODE REVIEW

File: Queens_Problem/Queens_Problem.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+				(*board)[i][j] = 1
+				if PlaceQueen(board, boardSize-1) {
+					return true
+				}
+				(*board)[i][j] = 0
+			}
+		}
+	}
+	return false
+}
```
Comment (ayushin78) [inline Queens_Problem/Queens_Problem.go]: Add input output example

File: Queens_Problem/Queens_Problem.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+					return true
+				}
+			}
+		}
+	}
+	return false
+}
+
+// PlaceQueen place queens on the board and will return true if it is possible to place all the queens else false
+func PlaceQueen(board *[][]int, boardSize int) bool {
```
Comment (ayushin78) [inline Queens_Problem/Queens_Problem.go]: use camelCase for function names. In golang, first letter is capitilized for exported identifiers.

File: Queens_Problem/Queens_Problem.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
+// Given a chess board having N×N cells place N queens on the board in such a way that no queen attacks any other queen.
+
+package main
+
+import "fmt"
+
+func main() {
+	boardSize := 8 // size of chess board (8 x 8)
+
```
Comment (ayushin78) [inline Queens_Problem/Queens_Problem.go]: Take boardSize as input from the user

File: Queens_Problem/Queens_Problem.go (hunk 1 above)
Comment (khushal87) [reply Queens_Problem/Queens_Problem.go]: added

**Comment by ayushin78** [comment]
Also add some comments in isAttacked function.  Specify when the row or column are being checked and when the diagonals are checked.

**Comment by khushal87** [comment]
All changes are done and the code is working @ayushin78

**Comment by ayushin78** [comment]
Looks good to me :+1: . Thanks @khushal87 :)

**Comment by khushal87** [comment]
> 
> 
> Looks good to me 👍 . Thanks @khushal87 :)

Thanks alot :)

**Pull request merged**